lib_eio/core/eio__core.mli

```diff
@@ -126,10 +126,10 @@ module Promise : sig
           (fun () -> Promise.resolve resolver 42)
       ]} *)
 
-  type !'a t
+  type +'a t
```

Comment:
```suggestion
  type +!'a t
```

Reply:
IIRC I added them then since compilation was failing in the case below:
> type _ eff += Fork : (unit -> 'a) -> 'a Promise.t eff

Does it not fail anymore? I believe we still need `a Promise.t` to be injective for it to be usable in the above use-case (i.e. non-existential GADT usage).